### ISSUE TITLE
Adding a new player to a league does not update the player list

### ISSUE THREAD

**Issue by makkoli** [freeCodeCamp/league-for-good#80]
Players tab > Manually adding a player to league

After successful insertion, the roster does not update with the new player.

**Comment by afranco07** [comment]
I want to try and work on this

**Comment by afranco07** [comment]
Hey I have this pretty much done, however I can't get the state to update the team roster without refreshing the page. Any help on which reducer to use for getting the latest team/league info?

**Comment by afranco07** [comment]
@IRGames yea I'm having trouble updating the state of the roster. There is a reducer `UPDATE_TEAM` under `teams.js` in the reducers folder, but I'm not sure if it is working correctly. And like I said, refreshing the page updates the state correctly

Here is some code I wrote attempting to update state. This is in the file `createPlayer.js` in this location `client/src/actions/players/` in the `creatPlayer(form, dispatch, props)` function, starts on line 31:

```javascript
if ( team && team.teamId) {
    const leagueID = data.leagueId;
    axios.get(`${ROOT_URL}/league/fetch/${leagueID}`)
        .then(leagueData => {
            dispatch({
                type: EDIT_LEAGUE,
                payload: {
                    [leagueID]: { ...leagueData.data }
                }
            });
        });
}
```

**Comment by IRGames** [comment]
When you try adding a player, does it also "break" the app, like in these pictures?
![image](https://user-images.githubusercontent.com/22033921/38908691-73bf967a-428f-11e8-86e4-a71933ee914a.png)
![image](https://user-images.githubusercontent.com/22033921/38908694-76b7ac64-428f-11e8-9400-41329e034418.png)

**Comment by afranco07** [comment]
@IRGames yes you are correct about the `data.pending = false;` and I already have some logic that checks if you are the admin for setting that. The problem is I need to somehow update the roster to include the new player you just created (without refreshing the page, just the state). If I go to the team roster, the player is not shown. However if I refresh the page then the player does show up in the roster. So there must be a way to get the state to update the roster, I just haven't found/figured it out yet.

Also it doesn't break like in the pictures you have above. There is an error that gets printed to the console but I don't exactly remember what it said. I'll be able to get that error for you tomorrow if you want. 

I did get that error in the pictures before while working on something else. It happened when I was sending the wrong payload into the Reducer. Double check the data you're sending in

**Comment by IRGames** [comment]
Ok!  If you could, I'd like to see the error.  When my app crashed on my end, no error appeared.

I noticed one of the "To Dos" for the app was to "Finish writing out roles" (admin, coach, etc).  When you get a chance, could you show me where you added the logic to check for admin?  I was wondering if/how the app is currently set up to check a user's role.  Thanks for talking things out with me here!

**Comment by afranco07** [comment]
@IRGames I new to react and redux as well so this might not be the best way to do it but, here's a little snippet on how I am getting the users role. I get the current users email from redux state and find them under the staff array in the redux state (I save it into user variable). Then I check if the users role is `Admin` using the ternary `if` statement.

```javascript
function mapStateToProps(state, ownProps) {
	// Getting the users role
	const email = state.auth.user.email;
	const staff = state.settings.staff;
	const user = staff.find(staffPerson => staffPerson.email === email);
	// cut out some code ...
	return {
		// removed some other code ...
		isAdmin: user.role === 'Administrator' ? true : false
	};
}
```

Then under createPlayer.js I have the line like you mentioned earlier:
```javascript
player.pending = isAdmin ? false : true;
```

And the error I get is `TypeError: can't assign to property "status" on "<insert long id number here>": not an object`  and it refers to the file `teamData.js` Line`156:5`

**Comment by IRGames** [comment]
@afranco07 Thanks for the reply!  When you've been looking at the app, have you seen any other places where a user's role has been checked?  I see that the Administrator, General Manager and Coach roles have been defined in `roles.config.js`, but I don't see any code that involves their implementation.  

I ask because of the following "To Do" in writing out the roles:
Finish writing out roles:
> Administrator - has access to all tabs; has access to all permissions
> 
> Manager - has access to teams, players, seasons tab; access to all permissions in those tabs
> 
> Coach - has access to teams tab; has permissions view team players, input team scores after games

If we're going to check to see if there's an administrator here, we'll want to stay consistent and use the same format for checking a user's role throughout the app.  Sounds like a big task.  Should we ask someone at freeCodeCamp if the two of us can go ahead and start working on this implementation?

**Comment by makkoli** [comment]
Hey guys, taking on the role task is a big ask. Significant parts of the app are in need of some refactoring to get it to work. You can join us on our slack at fcc-leagueforgood.slack.com. No one has been really using it, but I'm usually on slack everyday if you want to talk more about this.

**Comment by afranco07** [comment]
@makkoli how do we sign up for the slack?

**Comment by makkoli** [comment]
@afranco07 Yeah, I actually forgot I need to invite you to the workspace. If you give me your email I can send you an invite.

**Comment by afranco07** [comment]
@makkoli afranco527@gmail.com

**Comment by makkoli** [comment]
@afranco07 Sent.

**Comment by IRGames** [comment]
@makkoli isaac.w.raymond@gmail.com  
thanks!

**Comment by makkoli** [comment]
@IRGames sent

**Comment by paulywill** [comment]
@makkoli Can this issue be closed?

**Comment by afranco07** [comment]
@paulywill This issue has been fixed?
